factories_ZA/basePlotter.py

```diff
@@ -250,12 +249,12 @@ def generatePlots(self, categories, stage, requested_plots, weights, systematic=
                     llIdIso_sf_dict["err_lep2_el"] = "{id}[{0}][{1}] * {reco}[{0}][0]".format(self.lep2_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch)
 
                 if sf == "elreco": 
-                    llIdIso_sf_dict["err_lep1_el"] = "({lep1}.isEl && (({lep1}.p4.Pt() < 20.) || ({lep1}.p4.Pt() > 80.)})) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco[{0}][{0}]} * 0.01*{reco[{0}][{0}]})) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep1_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep1=self.lep1_str)
-                    llIdIso_sf_dict["err_lep2_el"] = "({lep2}.isEl && (({lep2}.p4.Pt() < 20.) || ({lep2}.p4.Pt() > 80.)})) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco[{0}][{0}]} * 0.01*{reco[{0}][{0}]})) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep2_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep2=self.lep2_str)
+                    llIdIso_sf_dict["err_lep1_el"] = "(({lep1}.p4.Pt() < 20.) || ({lep1}.p4.Pt() > 80.)) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco}[{0}][{0}] * 0.01*{reco}[{0}][{0}])) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep1_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep1=self.lep1_str)
+                    llIdIso_sf_dict["err_lep2_el"] = "(({lep2}.p4.Pt() < 20.) || ({lep2}.p4.Pt() > 80.)) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco}[{0}][{0}] * 0.01*{reco}[{0}][{0}])) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep2_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep2=self.lep2_str)
```

Comment:
why remove the `isEl` ?

Reply:
this check is not there for all the other SFs, not sure why I put it there. And the check on the flavor of the leptons is here already:

Reply:
Actually I'm thinking that the check should be there. The bug was the way the conditions are applied

Reply:
Ok, I remembered the bug. What the code was saying was:

if (lep1.isEl && (pT > ...)) {
    sf_uncertainty = unc1;
}
else {
    sf_uncertainty = unc2;
}

so in the else you're requiring the lepton to be a muon but with electron's SFs.
It should be instead:

if (lep1.isEl) {
    if ((pT > ...)) {
        sf_uncertainty = unc1;
    }
    else {
        sf_uncertainty = unc2;
    }
}